webapp/src/features/adminEditIntersection/adminEditIntersectionSlice.tsx

```diff
@@ -0,0 +1,234 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { selectToken } from '../../generalSlices/userSlice'
+import EnvironmentVars from '../../EnvironmentVars'
+import apiHelper from '../../apis/api-helper'
+import { updateTableData as updateIntersectionTableData } from '../adminIntersectionTab/adminIntersectionTabSlice'
+import { RootState } from '../../store'
+import { AdminEditIntersectionFormType } from './AdminEditIntersection'
+
+export type adminEditIntersectionData = {
+  intersection_data: AdminEditIntersectionFormType
+  allowed_selections: {
+    organizations: string[]
+    rsus: string[]
+  }
+}
+
+const initialState = {
+  apiData: {} as adminEditIntersectionData,
+  organizations: [] as { name: string }[],
+  selectedOrganizations: [] as { name: string }[],
+  rsus: [] as { name: string }[],
+  selectedRsus: [] as { name: string }[],
+  submitAttempt: false,
+}
+
+function isEmptyObject(obj: any): boolean {
+  return obj && Object.keys(obj).length === 0
+}
+
+export const checkForm = (state: RootState['adminEditIntersection']) => {
```

Comment:
Similar to previous comments, let's add some commenting into these methods to know what they are doing. "checkForm" seems to only be validating a property length is not 0, "updateJson" is removing null properties and updating some lists but without digging into the code we'd have no idea that's what it does.

Reply:
Added docstring comments and renamed method